### ISSUE TITLE
Malformed alert_json output for events with large b64_data fields values 

### ISSUE THREAD

**Issue by chriswhite199** [snort3/snort3#143]
I'm observing malformed json being output as part of alert_json usage, for example the servicefield has no associated value:

```
... "rule" : "1:1000020:1", "service" : , "sid" : 1000020, ...
```

It looks like the issue might be in the `TextLog_Quote` function (https://github.com/snort3/snort3/blob/master/src/log/text_log.cc#L292) where if writing the current field value would go beyond the max buffer size, the field write is aborted (return false), but nothing in `alert_json.cc` pays attention to the return value of the `TextLog_Quote` call, and this carries on.

I don't have a pcap file I can share to reproduce this, but wanted to raise the issue for comment anyways (based upon what I think I can see in the above code)

**Comment by chriswhite199** [comment]
I should note that this is usually observable when the b64_data field is quite large and would therefore dominate the LOG_BUFFER size

**Comment by mcampbell** [comment]
As a witness to this issue, I would add that the field that gets aborted is variable; in my case it was `dst_addr`.

**Comment by Xiche** [comment]
Ok, thank you very much for reporting this and digging in to debug it.  We'll take a look.

**Comment by Xiche** [comment]
Should be fixed in 159947458476b39588b771fba471f7fa9e8c8ca5 - thanks for reporting the issue!
